lib/Controller/SlaveController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * @copyright Copyright (c) 2017 Bjoern Schiessle <bjoern@schiessle.org>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+namespace OCA\GlobalSiteSelector\Controller;
+
+
+use Firebase\JWT\JWT;
+use OCA\GlobalSiteSelector\GlobalSiteSelector;
+use OCP\AppFramework\Http\RedirectResponse;
+use OCP\AppFramework\OCSController;
+use OCP\ILogger;
+use OCP\IRequest;
+use OCP\IURLGenerator;
+use OCP\IUserSession;
+
+require_once __DIR__ . '/../../vendor/autoload.php';
+
+/**
+ * Class SlaveController
+ *
+ * Endpoints in case the global site selector operates as a slave
+ *
+ * @package OCA\GlobalSiteSelector\Controller
+ */
+class SlaveController extends OCSController {
+
+	/** @var GlobalSiteSelector  */
+	private $gss;
+
+	/** @var ILogger */
+	private $logger;
+
+	/** @var IUserSession */
+	private $session;
+
+	/** @var IURLGenerator */
+	private $urlGenerator;
+
+	/**
+	 * SlaveController constructor.
+	 *
+	 * @param string $appName
+	 * @param IRequest $request
+	 * @param GlobalSiteSelector $gss
+	 * @param ILogger $logger
+	 * @param IUserSession $session
+	 * @param IURLGenerator $urlGenerator
+	 */
+	public function __construct($appName,
+								IRequest $request,
+								GlobalSiteSelector $gss,
+								ILogger $logger,
+								IUserSession $session,
+								IURLGenerator $urlGenerator
+	) {
+		parent::__construct($appName, $request);
+		$this->gss = $gss;
+		$this->logger = $logger;
+		$this->session = $session;
+		$this->urlGenerator = $urlGenerator;
+	}
+
+	/**
+	 * @PublicPage
+	 * @NoCSRFRequired
```

Comment:
This is kinda tricky because right now this would potentially allow login CSRF. This means I can make someone else login under an account of my control.

While not a super high security risk this is still considered bad by most security professionals as you could have a similar folder structure, the other person would then assume it's their account and store potential sensitive data on the account of the attacker. Let me think more about this :)